GibbsSampling.Library/PositionMatrix/PositionMatrixFunctions.fs

```diff
@@ -0,0 +1,137 @@
+    let internal checkForDistance (width:int) (items:int list) =
+        if items.Length <= 0 || items.Length = 1 then true
+        else
+            let rec loop n i =
+                if n = items.Length-1 then true
+                else
+                    if i >= items.Length then loop (n+1) (n+2)
+                    else
+                        if Operators.abs(items.[n]-items.[i]) > width then
+                            loop n (i+1)
+                        else false
+            loop 0 1
+
+    /// Get an integer which is between 0 and the length of the sequence - segmentLength
+    let internal getRandomNumberInSequence (segmentLength:int) (source:'a[]) =
+        let rnd = System.Random()
+        Array.init 1 (fun _ -> rnd.Next(0, source.Length-segmentLength+1))
+        |> Array.head
+
+    /// Create a specific sub sequence of the source sequence based on the given length and starting position. Do not forget, sequences start counting at 0!
+    let internal getSegment (subsequenceLength:int) (source:'a[]) (startPoint:int) =
+        source 
+        |> Array.skip startPoint 
+        |> Array.take subsequenceLength 
+        |> Array.ofSeq
+
+    ///Finds the best information content in an array of arrays of PWMSs and positions.
+    let getBestInformationContent (item:((float*int)[])[]) =
+        let rec loop (n:int) (bestPWMS:(float*int)[]) =
+            if n = item.Length then bestPWMS
+            else
+                let informationContentItem =
+                    item.[n]
+                    |> Array.fold (fun baseValue (pwms, _) -> pwms + baseValue) 0.
+                let informationContentBestPWMS =
+                    bestPWMS
+                    |> Array.fold (fun baseValue (pwms, _) -> pwms + baseValue) 0.
+                if informationContentItem > informationContentBestPWMS then
+                    loop (n + 1) item.[n]
+                else
+                    loop (n + 1) bestPWMS
+        loop 0 [||]
+
+    /// Increase counter of PositionFrequencyMatrix at fixed position by 1.
+    let increaseInPlacePFM (pos:int) (bioItem:'a when 'a :> IBioItem) (positionFrequencyMatrix:PositionFrequencyMatrix) = 
+        positionFrequencyMatrix.[bioItem, pos] <- positionFrequencyMatrix.[bioItem, pos] + 1
+        positionFrequencyMatrix
+
+    /// Increase counter of PositionFrequencyMatrix at fixed position by n.
+    let increaseInPlacePFMBy (pos:int) (bioItem:'a when 'a :> IBioItem) n (positionFrequencyMatrix:PositionFrequencyMatrix) = 
+        positionFrequencyMatrix.[bioItem, pos] <- positionFrequencyMatrix.[bioItem, pos] + n
+        positionFrequencyMatrix
+
+    /// Create PositionFrequencyMatrix based on BioArray.
+    let createPFMOf (source:BioArray.BioArray<#IBioItem>) =
+        let positionFrequencyMatrix = new PositionFrequencyMatrix(source.Length)
+        source
+        |> Array.fold (fun (row, cm) column -> row + 1, increaseInPlacePFM row column cm) (0, positionFrequencyMatrix) |> ignore
```

Comment:
why do you ignore the result of the fold? if you use an in place function, can't you use Array.iteri (i for row increments)?

Reply:
Also, your in place update returns the changed matrix instead of unit, so you could just return the result and not ignore it

Reply:
fixed